### DISCUSSION TITLE
Lorentz Fit Plot Offset

### DISCUSSION THREAD

**Discussion by framigo** [NanoComp/meep#2260] [General]
In [meep](https://github.com/NanoComp/meep)/[python](https://github.com/NanoComp/meep/tree/master/python)/[examples](https://github.com/NanoComp/meep/tree/master/python/examples)/eps_fit_lorentzian.py,

there is an offset of eps_inf between the actual and fit real(epsilon) curve:
![image](https://user-images.githubusercontent.com/1023314/194217358-d59fe5e7-3cf1-4b06-8c7b-62ba51d3272b.png)

I suppose this is because in 

https://github.com/NanoComp/meep/blob/40af0a3ac667ac2715eb769c97517f6d5367aeb1/python/examples/eps_fit_lorentzian.py#L146-L147

we are plotting the fitted epsilon against the actual epsilon-arbitrary eps_inf defined earlier:

https://github.com/NanoComp/meep/blob/40af0a3ac667ac2715eb769c97517f6d5367aeb1/python/examples/eps_fit_lorentzian.py#L80-L91

The fitting function itself works just fine.(?) Please help check on this.

**Comment by oskooi** [comment]
Fitting an arbitrary complex refractive index to a sum of Drude-Lorentzian resonances involves trial and error, unfortunately. In your case, it seems you just need to shift the real part down while keeping the imaginary part unchanged. Try *increasing* the `eps_inf` parameter gradually and see whether it improves the fit. In the example script, `eps_inf` is chosen arbitrarily as 1.1. `eps_inf` (the instantaneous dielectric at infinite frequency) is an important tuning parameter which must be set manually.

In my experience, for an index profile with a real part which is positive over the entire wavelength spectrum, obtaining a decent fit requires making the `eps_inf` parameter *as large as possible* ~~without making the real part negative~~ in order for the smallest real part of $\varepsilon$ over the bandwidth to be ~1.0. This point is already mentioned as a comment:

https://github.com/NanoComp/meep/blob/fba8aebb683620016d2f111f329cf8785039da2d/python/examples/eps_fit_lorentzian.py#L79

**Comment by framigo** [comment]
This happened consistently in many cases for me.

In fact, using eps_inf close to 0 works better than large eps_inf.

**Comment by htakamur** [comment]
This issue has been fixed by PRs (#2303, #2304).
